### PR TITLE
contracts: Add isWithdrawn() function to request manager

### LINKED ISSUE #1051: Create isWithdrawn() function for requests
As a developer I want to read the contract's state and see if a request is withdrawn or not. There is currently no exposed function for this.
Added value is, that developers who build on beamer can easily find out if a request is withdrawn or not. 

PSEUDO
```
function isWithdrawn(bytes32 requestId) public view returns (bool) {
    if(requests[requestId].withdrawnClaimId == 0) {
        return False;
    }
    return True;
```

- [ ] Frontend should be able to use this function

### PR DESCRIPTION
**Pull request by manuelwedler**
Closes #1051

This enables easily checking if the deposit for a request was withdrawn or not.
For the tests, I added calls to the new function in the withdrawal test functions, in order to not duplicate the whole test setup for this simple function. Only the withdrawal state for a new request got its own test.

### CODE REVIEW

**Comment by fredo** [comment]
could you also add an assert (or test) for testing that isWithdrawn is true after `withdrawExpiredRequest` executed by the requester

File: contracts/contracts/RequestManager.sol
Comment (istankovic) [inline contracts/contracts/RequestManager.sol:656]: Should we perhaps use the `validRequestId` modifier here?

**Comment by manuelwedler** [comment]
Thanks, just addressed the feedback

**Comment by istankovic** [comment]
@manuelwedler there are fixup commits, could you please squash them?

**Comment by manuelwedler** [comment]
Sure, will do so. Usually, I'm waiting for an approval before that.

**Review by istankovic — APPROVED** [review]
Thanks!

**Pull request merged**
